Description: Fix FTBFS on s390x
 error: ‘element’ may be used uninitialized in this function
 error: ‘cmpResult’ may be used uninitialized in this function
 error: ‘value’ may be used uninitialized in this function
 error: sprintf may write a terminating nul past the end of the destination
Author: Eduardo Barretto <eduardo.barretto@canonical.com>
Updated: 2024-03-21

--- a/nss/cmd/certutil/certext.c
+++ b/nss/cmd/certutil/certext.c
@@ -327,7 +327,7 @@ AddKeyUsage(void *extHandle, const char
     SECItem bitStringValue;
     unsigned char keyUsage = 0x0;
     char buffer[5];
-    int value;
+    int value = 0;
     char *nextPos = (char *)userSuppliedValue;
     PRBool isCriticalExt = PR_FALSE;
 
@@ -510,7 +510,7 @@ static SECStatus
 AddExtKeyUsage(void *extHandle, const char *userSuppliedValue)
 {
     char buffer[5];
-    int value;
+    int value = 0;
     CERTOidSequence *os;
     SECStatus rv;
     SECItem *item;
@@ -664,7 +664,7 @@ AddNscpCertType(void *extHandle, const c
     SECItem bitStringValue;
     unsigned char keyUsage = 0x0;
     char buffer[5];
-    int value;
+    int value = 0;
     char *nextPos = (char *)userSuppliedValue;
     PRBool isCriticalExt = PR_FALSE;
 
--- a/nss/lib/libpkix/pkix/util/pkix_list.c
+++ b/nss/lib/libpkix/pkix/util/pkix_list.c
@@ -1535,7 +1535,7 @@ PKIX_List_SetItem(
         PKIX_PL_Object *item,
         void *plContext)
 {
-        PKIX_List *element;
+        PKIX_List *element = NULL;
 
         PKIX_ENTER(LIST, "PKIX_List_SetItem");
         PKIX_NULLCHECK_ONE(list);
--- a/nss/lib/libpkix/pkix_pl_nss/system/pkix_pl_oid.c
+++ b/nss/lib/libpkix/pkix_pl_nss/system/pkix_pl_oid.c
@@ -107,7 +107,7 @@ pkix_pl_OID_Equals(
         PKIX_Boolean *pResult,
         void *plContext)
 {
-        PKIX_Int32 cmpResult;
+        PKIX_Int32 cmpResult = 0;
 
         PKIX_ENTER(OID, "pkix_pl_OID_Equals");
         PKIX_NULLCHECK_THREE(first, second, pResult);
